apis/refs/v1beta1/sqlinstanceref.go

```diff
@@ -22,3 +34,83 @@ type SQLInstanceRef struct {
 	/* The `namespace` field of a `SQLInstance` resource. */
 	Namespace string `json:"namespace,omitempty"`
 }
+
+type SQLInstance struct {
+	ProjectID       string
+	SQLInstanceName string
+}
+
+func (s *SQLInstance) String() string {
+	return "projects/" + s.ProjectID + "/instances/" + s.SQLInstanceName
+}
+
+func (r *SQLInstanceRef) ResolveRef(ctx context.Context, reader client.Reader, src client.Object) (*SQLInstance, error) {
```

Comment:
If we resolve to a SQLInstanceRef (with only external populated), then I don't think we need the second type, and I think we can resolve "in place" in the bigger object.  Is there a reason to prefer SQLInstance over SQLInstanceRef?

Reply:
I think I see what you mean. Let me try another attempt. Though this PR is a temp solution to unblock the BigQueryConnectionConnection object